common/persistence/visibility/store/sql/query_converter.go

```diff
@@ -138,6 +139,7 @@ func newQueryConverterInternal(
 func (c *QueryConverter) BuildSelectStmt(
 	pageSize int,
 	nextPageToken []byte,
+	withOrderBy bool,
```

Comment:
With the current design, there are limited options to avoid passing a control boolean here, which linter is also complaining about. 

IMO currently, select and count statement builders are tightly coupled with query converter without any direct dependencies which requires a lot of mocking and boiler-plating to unit-test its behavior. I think if we refactor the code and separate these based on their responsibilities, it makes it easier to reason, test, and pass parameters to them.